component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java

```diff
@@ -0,0 +1,481 @@
+    
+        String[] scopes;
+        try {
+            // If there was an error in the token info, abort.
+            if (tokenInfo.has("error")) {
+                throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR,
+                                         "Error during token validation: " + tokenInfo.get("error").toString());
+            }
+    
+            //depending of openid implementation
+            //issued_to can be named client_id
+            String clientId = tokenInfo.has(OAuthConstants.ISSUED_TO_PARAMETER) ?
+                              tokenInfo.getString(OAuthConstants.ISSUED_TO_PARAMETER) :
+                              tokenInfo.has(OAuthConstants.CLIENT_ID_PARAMETER) ?
+                              tokenInfo.getString(OAuthConstants.CLIENT_ID_PARAMETER) : null;
+            
+            if (clientId!=null && !clientId.equals(clientID)) {
+                throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR,
+                                         "Token's client ID does not match app's. clientID "
+                                             + "from tokenINFO: " + tokenInfo.get(OAuthConstants.ISSUED_TO_PARAMETER));
+            }
+    
+            if (log.isTraceEnabled()) {
+                log.trace("Successfully validated accessToken from openid : " + tokenInfo);
+            }
+            scopes = tokenInfo.has(OAuthConstants.SCOPE_PARAMETER) ? tokenInfo.getString(OAuthConstants.SCOPE_PARAMETER).split(" ") : null;
+        } catch (JSONException jsonException) {
+            throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR,
+                                     "Error during token validation: token format is ko");
+        }
+    
+        return new OpenIdAccessTokenContext(accessToken.getTokenData(), scopes);
+    
+    }
+    @Override
+    public <C> C getAuthorizedSocialApiObject(OpenIdAccessTokenContext accessToken, Class<C> socialApiObjectType) {
+        return null;
+    }
+    
+    @Override
+    public void saveAccessTokenAttributesToUserProfile(UserProfile userProfile, OAuthCodec codec, OpenIdAccessTokenContext accessToken) {
+        String encodedAccessToken = codec.encodeString(accessToken.accessToken.getAccessToken());
+        OAuthPersistenceUtils.saveLongAttribute(encodedAccessToken, userProfile, OAuthConstants.PROFILE_OPEN_ID_ACCESS_TOKEN,
+                                                false, chunkLength);
+        
+    }
+    
+    @Override
+    public OpenIdAccessTokenContext getAccessTokenFromUserProfile(UserProfile userProfile, OAuthCodec codec) {
+        String encodedAccessToken = OAuthPersistenceUtils.getLongAttribute(userProfile,
+                                                                           OAuthConstants.PROFILE_OPEN_ID_ACCESS_TOKEN, false);
+        String encodedAccessTokenSecret = OAuthPersistenceUtils.getLongAttribute(userProfile, OAuthConstants.PROFILE_LINKEDIN_ACCESS_TOKEN_SECRET, false);
```

Comment:
IMO PROFILE_OPEN_ID instead of PROFILE_LINKEDIN

Reply:
done